internal/services/batch/batch_pool_resource.go

```diff
@@ -1251,6 +1280,20 @@ func resourceBatchPoolRead(d *pluginsdk.ResourceData, meta interface{}) error {
 						osDiskPlacement = string(*config.OsDisk.EphemeralOSDiskSettings.Placement)
 					}
 					d.Set("os_disk_placement", osDiskPlacement)
+					if config.SecurityProfile != nil {
+						securityProfile := make([]interface{}, 0)
+						securityConfig := make(map[string]interface{})
+						securityConfig["host_encryption_enabled"] = pointer.ToBool(config.SecurityProfile.EncryptionAtHost)
+						if config.SecurityProfile.SecurityType != nil {
+							securityConfig["security_type"] = string(*config.SecurityProfile.SecurityType)
+						}
+						if config.SecurityProfile.UefiSettings != nil {
+							securityConfig["secure_boot_enabled"] = pointer.ToBool(config.SecurityProfile.UefiSettings.SecureBootEnabled)
+							securityConfig["vtpm_enabled"] = pointer.ToBool(config.SecurityProfile.UefiSettings.VTpmEnabled)
+						}
+						securityProfile = append(securityProfile, securityConfig)
+						d.Set("security_profile", securityProfile)
+					}
```

Comment:
should we move this into a flatten func?

Reply:
Okay, will do